### DISCUSSION TITLE
trivy.yaml file not working for certain fields, policy: in example.

### DISCUSSION THREAD

**Discussion by JC-19891989** [aquasecurity/trivy#4226] [Q&A] [answered]
## Description
Not all commands are getting picked up from the trivy.yaml file - the misconfiguration: part in particular

If I run trivy with the "trivy config --config-policy X ." options it picks up the policy folder or complains the directory does not exist. 

The fields misconfiguration:  policy: in the trivy yaml do not have the same results, the fields seem to be ignored. Some fields from misconfiguration: seem to work like include-non-failures:  but certain flags such as policy: do not. 

## What did you expect to happen?

if I use misconfiguration: policy: in the trivy.yaml file   that should map to  --config-policy from the terminal. Instead the command is ignored from the trivy.yaml file

## What happened instead?

The fields seem to be ignored, anything can be put in the fields, when set up correctly with correct paths, it does not pick up any additional rego files.

## Output of run with `-debug`:

Test run with pod file from https://kubernetes.io/docs/concepts/workloads/pods/

controlplane $ trivy config -d  --config-policy /THISDOESNOTEXIST .
2023-02-22T18:33:10.845Z        INFO    Loaded trivy.yaml
2023-02-22T18:33:10.846Z        DEBUG   Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-02-22T18:33:10.848Z        DEBUG   cache dir:  /root/.cache/trivy
2023-02-22T18:33:10.848Z        INFO    Misconfiguration scanning is enabled
2023-02-22T18:33:10.848Z        DEBUG   Policies successfully loaded from disk
2023-02-22T18:33:10.848Z        DEBUG   Walk the file tree rooted at '.' in parallel
2023-02-22T18:33:10.882Z        FATAL   filesystem scan error:
    github.com/aquasecurity/trivy/pkg/commands/artifact.Run
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:428
  - scan error:
    github.com/aquasecurity/trivy/pkg/commands/artifact.(*runner).scanArtifact
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:263
  - scan failed:
    github.com/aquasecurity/trivy/pkg/commands/artifact.scan
        /home/runner/work/trivy/trivy/pkg/commands/artifact/run.go:655
  - failed analysis:
    github.com/aquasecurity/trivy/pkg/scanner.Scanner.ScanArtifact
        /home/runner/work/trivy/trivy/pkg/scanner/scan.go:146
  - failed to call hooks:
    github.com/aquasecurity/trivy/pkg/fanal/artifact/local.Artifact.Inspect
        /home/runner/work/trivy/trivy/pkg/fanal/artifact/local/fs.go:160
  - post handler error:
    github.com/aquasecurity/trivy/pkg/fanal/handler.Manager.PostHandle
        /home/runner/work/trivy/trivy/pkg/fanal/handler/handler.go:75
  - misconfiguration scan error:
    github.com/aquasecurity/trivy/pkg/fanal/handler/misconf.misconfPostHandler.Handle
        /home/runner/work/trivy/trivy/pkg/fanal/handler/misconf/misconf.go:45
  - scan config error:
    github.com/aquasecurity/trivy/pkg/misconf.(*Scanner).Scan
        /home/runner/work/trivy/trivy/pkg/misconf/scanner.go:267
  - failed to load rego policies from [THISDOESNOTEXIST root/.cache/trivy/policy/content]: stat /THISDOESNOTEXIST: no such file or directory

create trivy.yaml with mirror config

misconfiguration:
  include-non-failures: false
  trace: false
  policy:
    - ./THISDOESNOTEXIST
  data:
    - data/

the job runs - no successful test show (as expected)

controlplane $ trivy config -d .
2023-02-22T18:35:17.411Z        INFO    Loaded trivy.yaml
2023-02-22T18:35:17.412Z        DEBUG   Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2023-02-22T18:35:17.412Z        DEBUG   cache dir:  /root/.cache/trivy
2023-02-22T18:35:17.412Z        INFO    Misconfiguration scanning is enabled
2023-02-22T18:35:17.412Z        DEBUG   Policies successfully loaded from disk
2023-02-22T18:35:17.413Z        DEBUG   Walk the file tree rooted at '.' in parallel
2023-02-22T18:35:18.339Z        DEBUG   OS is not detected.
2023-02-22T18:35:18.340Z        INFO    Detected config files: 1
2023-02-22T18:35:18.340Z        DEBUG   Scanned config file: pod.yaml

pod.yaml (kubernetes)

Tests: 141 (SUCCESSES: 129, FAILURES: 12, EXCEPTIONS: 0)
Failures: 12 (UNKNOWN: 0, LOW: 10, MEDIUM: 2, HIGH: 0, CRITICAL: 0)

MEDIUM: Container 'nginx' of Pod 'nginx' should set 'securityContext.allowPrivilegeEscalation' to false
═══════════════════════════════════════════════════════════════════════════════════════════════════════════
A program inside the container can elevate its own privileges and run as root, which might give the program control over the container and node.

See https://avd.aquasec.com/misconfig/ksv001

(etc)

When trivy.yaml is changed:

misconfiguration:
  include-non-failures: true

The output shows PASS results - so the config is working

PASS: HIGH: No issues found
═══════════════════════════════════════════════════════════════════════════════════════════════════════════
Storing secrets in configMaps is unsafe

See https://avd.aquasec.com/misconfig/avd-ksv-0109
───────────────────────────────────────────────────────────────────────────────────────────────────────────

PASS: MEDIUM: No issues found
═══════════════════════════════════════════════════════════════════════════════════════════════════════════
The RBAC role cluster-admin provides wide-ranging powers over the environment and should be used only where and when needed.

See https://avd.aquasec.com/misconfig/ksv111

etc

## Output of `trivy -v`:

controlplane $ trivy -v
2023-02-22T18:40:57.488Z        INFO    Loaded trivy.yaml
Version: 0.37.3
Vulnerability DB:
  Version: 1
  UpdatedAt: 2023-02-08 12:48:16.989777838 +0000 UTC
  NextUpdate: 2023-02-08 18:48:16.989777438 +0000 UTC
  DownloadedAt: 2023-02-22 18:19:02.973421723 +0000 UTC

**Comment by knqyf263** [comment, marked as answer]
Please reference the document.
https://aquasecurity.github.io/trivy/v0.41/docs/references/configuration/config-file/#rego-options
